src/Node.php

```diff
@@ -46,23 +48,29 @@ public function getStart() : int {
      */
     public function getFullStart() : int {
         foreach($this::CHILD_NAMES as $name) {
+
             if (($child = $this->$name) !== null) {
+
                 if (\is_array($child)) {
                     if(!isset($child[0])) {
                         continue;
                     }
                     $child = $child[0];
                 }
+
                 if ($child instanceof Node) {
                     return $child->getFullStart();
-                } elseif ($child instanceof Token) {
+                }
+
+                if ($child instanceof Token) {
                     return $child->fullStart;
                 }
+
                 throw new \Exception("Unknown type in AST: " . \gettype($child));
             }
         };
 
-        throw new \Exception("Unknown type in AST: " . \gettype($child));
+        throw new \RuntimeException("Could not resolve full start position");
```

Comment:
This exception assumed that `$child` was defined, and it seems misleading. The new message is broadly more accutate but lacks detail. In anycase I think this is a `Should never happen` exception.